src/main/java/longah/handler/PINHandler.java

```diff
@@ -13,34 +13,62 @@
 import java.util.logging.Level;
 
 /**
- * Handles the creation, loading, authentication and resetting of the PIN.
+ * Handles the creation, loading, authentication, and resetting of the PIN.
  */
 public class PINHandler {
-    private static Logger logger = Logger.getLogger("PIN Logger");
+    private static final Logger logger = Logger.getLogger("PIN Logger");
```

Comment:
have made the change in my side but it might cause conflict, ill handle it, tldr is use the general logger class if necessary, exceptions are by default logged

Reply:
same thing for any ui calls including print/read